src/lib.rs

```diff
@@ -169,7 +169,7 @@ pub async fn bootstap(mut shutdown: broadcast::Receiver<()>, config: Config) ->
                 received_notifications: notification_counter,
             },
         )
-    } else {
+    } else if state.config.environment != "test" {
```

Comment:
not needed please use `cfg!`

Reply:
How?... 🙏

Reply:
See `./providers/mod.rs` for examples using the `#[cfg()]` or here are some doc links: https://doc.rust-lang.org/rust-by-example/attribute/cfg.html, https://doc.rust-lang.org/reference/conditional-compilation.html